geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -11,30 +11,94 @@ package org.geomesa.nifi.datastore.processor
 
 import java.io.InputStream
 
+import org.apache.nifi.components.PropertyDescriptor
 import org.apache.nifi.flowfile.FlowFile
 import org.apache.nifi.processor._
 import org.apache.nifi.processor.io.InputStreamCallback
+import org.geomesa.nifi.datastore.processor.AbstractGeoIngestProcessor.checkCompatibleSchema
+import org.geomesa.nifi.datastore.processor.AvroIngestProcessor.{AvroMatchMode, buildWriter}
 import org.geotools.data._
+import org.geotools.feature.simple.SimpleFeatureBuilder
+import org.geotools.util.Converters
 import org.locationtech.geomesa.features.avro.AvroDataFileReader
 import org.locationtech.geomesa.utils.geotools._
-import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
+import scala.util.{Failure, Success}
 import scala.util.control.NonFatal
 
+object AvroIngestProcessor {
+  val AvroMatchMode: PropertyDescriptor =
+    new PropertyDescriptor.Builder()
+      .name("Avro SFT match mode")
+      .description("Determines how Avro SFT mismatches are handled")
+      .required(false)
+      .defaultValue("strict")
+      .allowableValues("strict", "lenient")
+      .build()
+
+  def buildWriter(sft1: SimpleFeatureType,
+                  sft2: SimpleFeatureType,
+                  matchMode: String): (FeatureWriter[SimpleFeatureType, SimpleFeature], SimpleFeature) => Unit = {
+    checkCompatibleSchema(sft1, sft2) match {
+      case Success(true) =>
+        // Schemas match, so use a regular writer.
+        // TODO:  Discuss if the useProvidedFid should be true!
+        (fw: FeatureWriter[SimpleFeatureType, SimpleFeature], sf: SimpleFeature) =>
+          FeatureUtils.write(fw, sf, true)
+      case Failure(error) =>
+        if (matchMode == "lenient") {
```

Comment:
would probably be good to make "lenient" and "strict" constant vals in the companion object